### DISCUSSION TITLE
Wait a certain time before calling the send_hook for the same item

### DISCUSSION THREAD

**Discussion by Dilolcyy** [Rapptz/discord.py#6847] [Q&A]
My program checks a number of items via URLs and calls a send_hook, when the item is in stock. I would like my program to wait a certain time before calling the send_hook for the same item again. It should however continue to check the other URLs. .. Is this possible?

Here is my program :

```
from bs4 import BeautifulSoup
from discord import send_hook
from datetime import datetime

pid = ['PB00393363','PB00430691','PB00430699']

headers = {
    'user-agent' : 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/90.0.4430.85 Safari/537.36'
}

while True:
    for i in range(len(pid)):
```
    
    url = "https://www.ldlc.com/fr-be/fiche/{}.html".format(pid[i])

    response = requests.get(url, headers=headers)     
    soup = BeautifulSoup(response.text, 'html.parser')
    soup = BeautifulSoup(requests.get(url).content, 'html.parser')
    stock_statut = soup.find('div', class_='content').get_text().strip()            
    product_title = soup.find('h1').get_text().strip()
    
    price = soup.find_all('div', class_='price')[3].get_text()
    

    try:

        image_url = soup.find('img', {'id':'ctl00_cphMainContent_ImgProduct'})['src']
    
    except:
       
        image_url = soup.find('img', {'id':'ctl00_cphMainContent_ImgProduct'})['src']
        
    if stock_statut == 'En stock':
        print('[{}] En stock'.format(str(datetime.now())))
        
        send_hook(product_title, url, image_url, price)

    else:
        print('[{}] Rupture'.format(str(datetime.now())))`
        
        
If you could help me that will make me happy. Thank you!

**Comment by apple502j** [comment]
`send_hook` is not a discord.py feature?

**Reply by Dilolcyy** [reply]
Yes, exactly. My discord.py code :
```
from discord_webhook import DiscordWebhook, DiscordEmbed

webhook = DiscordWebhook(url='https://discord.com/api/webhooks/83....', username="....")

def send_hook (product_title, url, image_url, price):

    embed = DiscordEmbed(title=product_title, url=url, description='EN STOCK!', color='03b2f8')
    embed.set_author(name='provided by TAS')
    #embed.set_footer(text='Embed Footer Text')
    embed.set_timestamp()
    embed.add_embed_field(name='Prix', value=price)
    embed.set_thumbnail(url=image_url)
    embed.set_footer(icon_url='https://media.ldlc.com/cms/2021-02/LDLC-logo_1.gif', text='Vendu par LDLC')
    #eembed.add_embed_field(name='Field 2', value='dolor sit')
    #embed.add_embed_field(name='Field 3', value='amet consetetur')
    #embed.add_embed_field(name='Field 4', value='sadipscing elitr')

    webhook.add_embed(embed)
    response = webhook.execute()
    webhook.remove_embed(0)
```
